app/src/main/java/nu/yona/app/api/manager/impl/NotificationManagerImpl.java

```diff
@@ -162,82 +165,40 @@ public void onError(Object errorMessage) {
         }
     }
 
+    public YonaMessages processYonaMessages(YonaMessages result){
+        YonaMessages resultYonaMessagesObj = (YonaMessages) result;
+        Embedded embedded = resultYonaMessagesObj.getEmbedded();
+        List<YonaMessage> listMessages = embedded.getYonaMessages();
+        SimpleDateFormat sdf = new SimpleDateFormat(AppConstant.YONA_LONG_DATE_FORMAT, Locale.getDefault());
+        for (YonaMessage message : listMessages) {
+            String messageEnumType = (message.getStatus() != null)?message.getStatus():(message.getDropBuddyReason()!=null)?message.getDropBuddyReason():message.getChange();
+            message.setNotificationMessageEnum(NotificationMessageEnum.getNotificationMessageEnum(message.getType(), messageEnumType));
+            if (message.getLinks() != null && message.getLinks().getYonaUser() != null && !TextUtils.isEmpty(message.getLinks().getYonaUser().getHref())) {
+                if (message.getEmbedded() == null) {
+                    message.setEmbedded(new Embedded());
+                }
+                message.getEmbedded().setYonaUser(getYonaUser(message.getLinks().getYonaUser().getHref()));
+            }
+            String uploadDate = "";
+            String createdTime = message.getCreationTime();
```

Comment:
Move this line and the one above inside the try/catch-block and then extract the entire try/catch-block into a separate method. Move the creation of the ``SimpleDateFormat`` into that method too.